extensions/pyRevitTools.extension/pyRevit.tab/Modify.panel/edit2.stack/Flip.pushbutton/script.py

```diff
@@ -1,25 +1,49 @@
 """Provides options for flipping (Hand/Face) selected elements."""
 
-from pyrevit import revit, DB
+from pyrevit import revit, DB, script
 from pyrevit import forms
 
 
 __context__ = 'selection'
 
+def can_flip_wall(el):
+    return isinstance(el, DB.Wall)
+
+
+def can_flip_facing(el):
+    return (hasattr(el, 'CanFlipFacing') and el.CanFlipFacing) or hasattr(el, 'rotate')
+
+
+def can_flip_hand(el):
+    return hasattr(el, 'CanFlipHand') and el.CanFlipHand
+
+
+def can_flip(el):
+    return hasattr(el, 'Flip')
+
+
+def flip():
+    with revit.Transaction('Flip Selected'):
+        for el in revit.get_selection():
+            if hasattr(el, 'Flip'):
+                el.Flip()
+
 
 def flip_facing():
     with revit.Transaction('Flip Facing Selected'):
         for el in revit.get_selection():
-            if hasattr(el, 'flipFacing'):
+            if hasattr(el, 'CanFlipFacing') and el.CanFlipFacing:
                 el.flipFacing()
-            elif hasattr(el, 'Flip'):
+            elif can_flip(el):
                 el.Flip()
+            elif hasattr(el, 'rotate'):
```

Comment:
If FlipFacing is not available - rotate the family to 180 Degrees (works the same for Doors, Windows, etc.)